base/abstractdict.jl

```diff
@@ -522,6 +522,9 @@ function ==(l::AbstractDict, r::AbstractDict)
     return anymissing ? missing : true
 end
 
+# Fallback implementation
+sizehint!(s::AbstractDict, n) = nothing
```

Comment:
```suggestion
sizehint!(s::AbstractDict, n) = s
```

Reply:
If so, should it be changed also for `AbstractSet`?
https://github.com/JuliaLang/julia/blob/1b686b2633052e21760cc936abe3a54834927948/base/abstractset.jl#L4

Reply:
Hmm. Yea, I would expect the collection to be returned like most APIs like this.